lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -720,3 +727,67 @@ hex = convertToBase Base16
 -- | Decode a 'ByteString' from base16
 fromHex :: ByteArray bout => ByteString -> Either String bout
 fromHex = convertFromBase Base16
+
+data ErrUnXPrvStripPub
+    = ErrCannotRoundtrip
+      -- ^ The resulting bytestring would have been unable to roundtrip using
+      -- @xPrvFromStrippedPubXPrv@. Most likely because the input @XPrv@ was
+      -- encrypted, or because it was an old (Byron) key.
+    deriving (Eq, Show)
+
+-- | Convert a @XPrv@ to a 96-byte long extended private key that does /not/
+-- include the public key.
+--
+-- The format is:
+-- > Extended Private Key (64 bytes) <> ChainCode (32 bytes)
+--
+-- Returns @Left@ if the resulting bytestring fails to roundtrip back to the
+-- original @XPrv@. This can happen:
+-- - If the @XPrv@ was encrypted
+-- - If a DerivationScheme1 (Byron) key was used (that does not conform to the
+-- "tweak")
+unXPrvStripPub :: XPrv -> Either ErrUnXPrvStripPub ByteString
+unXPrvStripPub k = do
+    let res = stripPub . unXPrv $ k
+
+    -- Check that it roundtrips.
+    case (fmap unXPrv . xPrvFromStrippedPubXPrv $ res) of
+        Right bytes
+            | bytes == unXPrv k -> Right res
+            | otherwise         -> Left ErrCannotRoundtrip
+        Left  _                 -> error "unXPrvStripPub: this state cannot be \
+            \reached from a rightfully crafted XPrv"
+  where
+    -- Converts  xprv <> pub <> cc
+    -- To        xprv <>        cc
```

Comment:
minor note here, `xprv` should really be `prv`. The`xprv` is so-to-speak `prv` + `cc`

Reply:
I changed this after reading your notes (was slightly wondering about it before), which use the term:

> 64 bytes: an extended Ed25519 secret key composed of:
>  - 32 bytes: Ed25519 curve scalar from which few bits have been tweaked (see below)
>  - 32 bytes: Ed25519 binary blob used as IV for signing